src/applications/vaos/actions/newAppointment.js

```diff
@@ -263,30 +273,34 @@ export function openFacilityPage(page, uiSchema, schema) {
     let eligibilityData = null;
     let parentId = newAppointment.data.vaParent;
     let facilityId = newAppointment.data.vaFacility;
+    let rootOrg = null;
 
     try {
       // If we have the VA parent in our state, we don't need to
       // fetch them again
       if (!parentFacilities) {
-        parentFacilities = await getParentFacilities(userSystemIds);
+        parentFacilities = await getOrganizations(userSystemIds);
       }
 
       const canShowFacilities = !!parentId || parentFacilities?.length === 1;
 
       if (canShowFacilities && !parentId) {
-        parentId = parentFacilities[0].institutionCode;
+        parentId = parentFacilities[0].id;
+      }
+
+      if (parentId) {
+        rootOrg = getRootOrganization(parentFacilities, parentId);
       }
 
-      const systemId = parentFacilities?.find(
-        parent => parent.institutionCode === parentId,
-      )?.rootStationCode;
       facilities =
         newAppointment.facilities[`${typeOfCareId}_${parentId}`] || null;
 
       if (canShowFacilities && !facilities) {
         facilities = await getFacilitiesBySystemAndTypeOfCare(
-          systemId,
-          parentId,
+          // Remove parse function when converting this call to FHIR service
+          parseFakeFHIRId(rootOrg.id),
```

Comment:
The fact that I had to add this in several places I think means that we might want to start from the end of the form and work backwards when making these updates. So, slots > clinics > facilities.